src/interface_adapter/getFilter/GetFilterPresenter.java

```diff
@@ -0,0 +1,38 @@
+package interface_adapter.getFilter;
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.ViewModel;
+import interface_adapter.search.SearchState;
+import interface_adapter.search.SearchViewModel;
+import use_case.getFilter.GetFilterOutputBoundary;
+import use_case.getFilter.GetFilterOutputData;
+import view.GetFilterView;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+
+public class GetFilterPresenter implements GetFilterOutputBoundary {
+
+    private final GetFilterViewModel getFilterViewModel;
+    private final ViewManagerModel viewManagerModel;
+
+    public GetFilterPresenter(GetFilterViewModel getFilterViewModel, ViewManagerModel viewManagerModel) {
+        this.getFilterViewModel = getFilterViewModel;
+        this.viewManagerModel = viewManagerModel;
+    }
+
+    @Override
+    public void prepareView(GetFilterOutputData filter) {
```

Comment:
Why do we only prepare view and decide in the view stage based on Notice. This should be changed to preparing success view and fail views so that the decision making is not left to the vies, rather the interface adapters

Reply:
Resolved. As the use case is mainly UI, the decision making has to happen in the view